codalab/apps/web/templates/web/admin_competitions_manager.html

```diff
@@ -4,154 +4,336 @@
+                            throw new Error('Something went wrong');
+                        })
+                        .then(json => {
+                            this.competitionsTableItems = json;
+                        })
+                        .catch(error => {
+                            console.log("error while fetching /api/admin/competitions", error);
+                        })
+                        .finally(() => {
+                            this.competitionsTableLoading = false;
+                        });
+                    },
+                    save(item) {
+                        // Mark the competition as competition to send through the API
+                        var newCompIdToUpdate = [item.id];
 
-        // Init the array of selected competitions
-        for (let i = 0; i < competitionCheckboxes.length; i++) {
-            if (competitionCheckboxes[i].checked) {
-                selectOneClicked(competitionCheckboxes[i]);
-            }
-        }
+                        // Propagate the modification to all selected rows and mark them as competitions to send through the API
+                        for (const selectedCompetition of this.competitionsTableSelected) {
+                            var competition = this.competitionsTableItems.find(competition => competition.id == selectedCompetition.id);
+                            competition.upper_bound_max_submission_size = item.upper_bound_max_submission_size;
+                            newCompIdToUpdate.push(selectedCompetition.id);
+                        }
 
-        // Add onChange event listener to inputs
-        for (let input of competitionUpperBoundMaxSubmissionSizeInputs) {
-            input.addEventListener('change', () => {
-                maxSubmissionSizeChanged(input);
+                        // Add them to already existing marked competitions
+                        this.competitionsIdToUpdate = [...new Set([...this.competitionsIdToUpdate, ...newCompIdToUpdate])];
+                    },
+                    saveAll() {
+                        var competitionsUpdated = [];
+                        for (const id of this.competitionsIdToUpdate) {
+                            const competition = this.competitionsTableItems.find(competition => competition.id == id);
+                            competitionsUpdated.push({
+                                id: id,
+                                upper_bound_max_submission_size: competition.upper_bound_max_submission_size
+                            });
+                        }
+                        if (competitionsUpdated.length > 0) {
+                            const competitionsJson = {
+                                competitions: competitionsUpdated
+                            }
+                            this.saving = true;
+                            fetch("/api/admin/competitions/update", {
+                                method: 'POST',
```

Comment:
You mean saveAll() uses a POST method and synchronize() uses a PATCH mathod.
I am not 100% aware of the differences between POST PUT and PATCH (there are also a lots of debate around them).
But I am used to use a POST or PUT method when I want to create/update objects based on the body sent along with the request.
So saveAll() seems correct to me.
On the other hand, the synchronize() method does not send any body, because it is not needed. This one is tricky because I am just asking the server to act on a resource based on the url and to update it based on another resource in the DB, which I am not giving in my request. I don't believe there is a method for that kind of behavior.
So I just chose one that does not feel too weird (meaning not POST and PUT, at least in my mind).
But may be I am wrong.
It is open to discussion